File: runtime/go/units/wasi/environ_sizes_get.go
// requires: memory/i32_store
func (w *WASI) wasi_environ_sizes_get(countPtr, bufSizePtr uint32) uint32 {
    total := uint32(0)
    for _, e := range w.env {
        total += uint32(len(e)) + 1
    }
    w.memory.i32_store(uint64(countPtr), uint32(len(w.env)))
    w.memory.i32_store(uint64(bufSizePtr), total)
    return wasiOk
}

File: crates/dewasm-backend-java/units/wasi/environ_sizes_get.java
// requires: memory/i32_store
int wasi_environ_sizes_get(int countPtr, int bufSizePtr) {
    int total = 0;
    for (byte[] e : env) {
        total += e.length + 1;
    }
    memory.i32_store(Integer.toUnsignedLong(countPtr), env.length);
    memory.i32_store(Integer.toUnsignedLong(bufSizePtr), total);
    return WASI_OK;
}
